controllers/atlasmigration_controller_test.go

```diff
@@ -295,12 +295,32 @@ func TestReconcile_extractCloudMigrationData(t *testing.T) {
 	cleanUp()
 }
 
-func TestReconcile_createTmpDir(t *testing.T) {
+func TestReconcile_createTmpDirFromMap(t *testing.T) {
+	tt := newMigrationTest(t)
+
+	// When the configmap exists
+	dir, cleanUp, err := tt.r.createTmpDirFromMap(context.Background(), map[string]string{
```

Comment:
I prefer not to write unit tests from private functions. They aren't part of the API and testing them directly reduces flexibility. Find a way to test the behavior using the components exposed API.

Reply:
I have added more tests